application.py

```diff
@@ -1,13 +1,478 @@
+        developed_company = developed["partnership_company"]
+
+    data = {
+        "user": [u for u in user],
+        "details": [
+            {
+                "name": project["project_name"],
+                "short_name": project["project_short_name"],
+                "documentation_link": [project["doc_link"]],
+                "project_description": project["project_description"],
+            }
+        ],
+        "developed": [developed["developed"], [developed_company]],
+        "source_control": source_control,
+        "architecture": {
+            "hosting": hosting,
+            "database": database,
+            "languages": languages,
+            "frameworks": frameworks,
+            "CICD": integrations,
+            "infrastructure": infrastructure,
+        },
+        "stage": stage,
+    }
+    print(data)
+    try:
+        projects = requests.post(
+            f"{API_URL}/api/projects",
+            json=data,
+            headers=headers,
+        )
+    except Exception:
+        try:
+            print(f"Request was not blocked but returned: {projects.json()}")
+        except Exception as second_error:
+            print(f"{second_error.__class__.__name__}: {second_error}")
+
+    return redirect(url_for("dashboard"))
+
+
+# ------------------------
+# PRE-SURVEY RENDERING
+# ------------------------
+
+
+@app.route("/pre-survey", methods=["GET"])
+def pre_survey():
+    return render_template("/pre_survey/pre-survey.html")
+
+
+@app.route("/pre-survey/project", methods=["GET"])
+def project_pre_survey():
+    return render_template("/pre_survey/pre-survey-project.html")
+
+
+@app.route("/pre-survey/architecture", methods=["GET"])
+def architecture_pre_survey():
+    return render_template("/pre_survey/pre-survey-architecture.html")
+
+
+@app.route("/pre-survey/technology", methods=["GET"])
+def tech_pre_survey():
+    return render_template("/pre_survey/pre-survey-tech.html")
+
+
+# ------------------------
+# PROJECT SECTION RENDERING
+# ------------------------
+
+
+@app.route("/survey/contact_tech", methods=["GET"])
+def contact_tech():
+    return render_template("/section_project/contact_tech.html")
+
+
+@app.route("/survey/contact_manager", methods=["GET"])
+def contact_manager():
+    return render_template("/section_project/contact_manager.html")
+
+
+@app.route("/survey/project", methods=["GET"])
+def project():
+    return render_template("/section_project/project.html")
+
+
+@app.route("/survey/stage", methods=["GET"])
+def stage():
+    return render_template("/section_project/stage.html")
+
+
+@app.route("/survey/developed", methods=["GET"])
+def developed():
+    return render_template("/section_project/developed.html")
+
+
+# ------------------------
+# CODE AND ARCHITECTURE SECTION RENDERING
+# ------------------------
+
+
+@app.route("/survey/source_control", methods=["GET"])
+def source_control():
+    stage = request.args.get("stage")
+    if stage == "2":
```

Comment:
What is stage == 2?  Could you introduce a meaningful enum for this?

Reply:
Changed the 2 functions with 'stages' to use a enums.